### ISSUE TITLE
A11y Review #2 WCAG Items

### ISSUE THREAD

**Issue by CalvinRodo** [cds-snc/c19-benefits-node#347]
## Major:

- [x] ~~WCAG 2.0: When the “Report a problem or mistake on this page” summary receives keyboard focus, no focus outline appears around it. I’d recommend adding a focus outline to it.~~
- [x] ~~WCAG 2.0: “Report a problem or mistake on this page” is coded as a summary for a `< details>` element, but doesn’t look like one. I’d recommend adding a triangle just before the text it to make it obvious that it’s expandable/collapsible content (i.e. a triangle that points rightwards when collapsed and changes to point downwards when expanded).~~
- [ ] WCAG 2.0: There’s no semantic markup in use to programmatically-indicate the importance of the legends’ red content (e.g. red wildcard, “(required)” and error message text). To address this, I’d recommend placing the red content within `<strong>` elements.
- [ ] WCAG 2.0: In the results page, the red box that appears if any questions were missed is visually-highlighted to make it look more important than the intro paragraph. But it’s coded like a normal paragraph. So screen reader users have no means of telling it’s more important. To help distinguish it, I’d recommend either using a `<strong>` element to add a bolded prefix to the box’s content (e.g. “Error: You missed…”) or wrapping a `<strong>` element around all of the box’s content (doesn’t necessarily have to look bolded though).

##  Medium:
- [x] ~~WCAG 2.0: The Your reduced income page’s page title doesn’t clearly identify its purpose. Its H1 is called “Your reduced income”, but its page title is “Find financial help during COVID-19”. None of the other pages have that issue, so maybe this page was missed when the other page titles were changed to use their H1 content as prefixes in their page titles? I’d recommend changing that page’s title to “Your reduced income — Find financial help during COVID-19”.~~
- [ ] WCAG 2.0: Form validation error messages use an H3 heading called “Please correct the errors on the page”. But an H3 isn’t appropriate within that context. I’d recommend changing that heading into an H2.
- [ ] WCAG 2.0: The results page’s “Other potential help” section contains a box that starts out with an H2 heading that’s called “More money through the GST credit”. Since the box is presented as a subsection, it should use a lower level heading . Therefore, I’d recommend changing “More money through the GST credit” into an H3.
- [ ] WCAG 2.1: The darkened orange outline is much better than before, but still has issues with green buttons. The orange outline doesn’t achieve a 3:1 contrast against green buttons that have focus. The green buttons change to dark green when they have focus, but the contrast difference between normal and dark green is still less than 3:1. To resolve this, I’d recommend adding a small amount of white space between the green buttons and their orange outlines. That’ll make the green buttons’ focus outlines obvious to all sighted users.
- [x] ~~WCAG 2.1: “Report a problem on this page” is visually-presented as a button. But in order to perceive it as a button, users need to be able to make out its light grey background colour. Users with bad vision will just perceive it as a random line of text. In order to comply with WCAG 2.1’s non-text contrast success criterion, at least one of the following would need to be done:
  - Add a triangle just before “Report a problem on this page” to make it obvious it’s expandable/collapsible content. Doing just this would be enough to pass both WCAG 2.0 and 2.1. The triangle would make it obvious to everyone that it’s a <summary>. As a result, the light grey background colour could be kept as-is since it would no longer be anything more than a visual decoration (WCAG 2.1 doesn’t care about visual decorations).
  - Consider adding a dark border around the button shape. The border’s contrast ratio against white would need to be at least 3:1. A border would make it obvious to all sighted users that they’re looking at some kind of button. This can be skipped if a triangle is added. Could still be a nice-to-have though.~~

## Minor:

- [ ] WCAG 2.0: In the results page’s “More money through the GST credit” section, “On average:” is followed by 2 lines that look like list content (“$400 for single people” and “$600 for couples”). But they’re not coded as a real list. I’d recommend coding them as an unordered list (`<ul>`).

**Comment by CalvinRodo** [comment]
https://github.com/cds-snc/c19-benefits-node/issues/344 fixes two of the items identified

**Comment by CalvinRodo** [comment]
Going to close this as they are all tracked as seperate issues.
